pkg/cloudproviders/aws/cloudprovider/cloudprovider.go

```diff
@@ -293,6 +293,9 @@ func getCABundle(ctx context.Context) *string {
 }
 
 func kubeDNSIP(ctx context.Context, clientSet *kubernetes.Clientset) (net.IP, error) {
+	if clientSet == nil {
+		return nil, fmt.Errorf("no K8s client provided")
+	}
```

Comment:
I'd lean away from nil checking for interfaces that we have full control over and don't expose as a core library. This pattern leads us down a road of checking all nilable parameters in every method, which I don't think is sustainable.